### PR TITLE
JS orchestrator for integration testing and cleanup

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**
- Write JS orchestrator for integration tests
- Write integration test
- Format
- Cleanup
- Refactor some code into utils with their own tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +60,15 @@
     "sonar-runner"
   ],
   "scripts": {
-    "test": "npx nyc mocha",
-    "ci": "npm run test && node ci-analysis.js"
+    "test-unit": "npx nyc mocha test/unit/**/*.test.js",
+    "test-integration": "npx nyc mocha test/integration/**/*.test.js",
+    "ci": "npm run test && node ci-analysis.js",
```
Comment (ilia-kebets-sonarsource) [inline package.json:65]: keep this

File: package.json [hunk 2]
```diff
@@ -56,7 +60,15 @@
...
+    "format": "prettier --write \"{src,specs}/**/!(*.lint).js\""
```
Comment (ilia-kebets-sonarsource) [inline package.json:66]: add assertions script

Comment (ilia-kebets-sonarsource) [reply package.json:66]: build script which runs everything, executed in the ci. should fail if formatting fails

Comment (ilia-kebets-sonarsource) [reply package.json:66]: do this outside of this PR

File: test/integration/scanner.test.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        console.log(error);
+      }
+    });
+    after(function () {
+      this.timeout(10 * 1000);
+      stop(sqPath);
+    });
+    it.only('should run an analysis', async function () {
+      await scannerPromise({
+        serverUrl: 'http://localhost:9000',
```
Comment (ilia-kebets-sonarsource) [inline test/integration/scanner.test.js:35]: we could generate a random port

File: tools/orchestrator/src/sonarqube.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+async function waitForStart() {
+  let isReady = false;
+  while (! isReady) {
+    try {
+      const [response] = await Promise.all([
+        isApiReady(),
+        sleep(),
+      ]);
+      isReady = response.data;
+      console.log('server ready')
```
Comment (ilia-kebets-sonarsource) [inline tools/orchestrator/src/sonarqube.ts:82]: remove

File: tools/orchestrator/src/sonarqube.ts [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+}
+
+/**
+ * Stop SonarQube instance
+ *
+ * @param sqPath The path where SQ was downloaded and unzipped
+ * @returns
+ */
+export function stop(sqPath: string = DEFAULT_FOLDER) {
+  const pathToBin = path.join(sqPath, 'bin', 'macosx-universal-64', 'sonar.sh');
```
Comment (ilia-kebets-sonarsource) [inline tools/orchestrator/src/sonarqube.ts:100]: change path here for OS

File: tools/orchestrator/src/sonarqube.ts [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+  });
+  return response.data.issues;
+}
+
+/**
+ * Non working generateToken() without using axios - to fix or erase
+ *
+ * @returns
+ */
+export function generateToken2(): Promise<any> {
```
Comment (ilia-kebets-sonarsource) [inline tools/orchestrator/src/sonarqube.ts:156]: remove this

File: tools/orchestrator/src/sonarqube.ts [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+/**
+ * Generate random alphanumeric string of given length (+1).
+ * We append '1' at the end to ensure we have at least 1 number,
+ * because it's required for the projectKey
+ *
+ * @param length
+ * @returns the random string
+ */
+function generateId(length: number = 10): string {
+  var result           = '';
```
Comment (ilia-kebets-sonarsource) [inline tools/orchestrator/src/sonarqube.ts:194]: use let

File: tools/orchestrator/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    // "disableSourceOfProjectReferenceRedirect": true,  /* Disable preferring source files instead of declaration files when referencing composite projects. */
+    // "disableSolutionSearching": true,                 /* Opt a project out of multi-project reference checking when editing. */
+    // "disableReferencedProjectLoad": true,             /* Reduce the number of projects loaded automatically by TypeScript. */
+
+    /* Language and Environment */
+    "target": "es2016",                                  /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
+    // "lib": [],                                        /* Specify a set of bundled library declaration files that describe the target runtime environment. */
+    // "jsx": "preserve",                                /* Specify what JSX code is generated. */
+    // "experimentalDecorators": true,                   /* Enable experimental support for TC39 stage 2 draft decorators. */
+    // "emitDecoratorMetadata": true,                    /* Emit design-type metadata for decorated declarations in source files. */
```
Comment (ilia-kebets-sonarsource) [inline tools/orchestrator/tsconfig.json:18]: cleanup all commented options

File: package.json (hunk 1 above)
Comment (ilia-kebets-sonarsource) [reply package.json:65]: rename unit-test as test

File: test/integration/scanner.test.js (hunk 3 above)
Comment (ilia-kebets-sonarsource) [reply test/integration/scanner.test.js:35]: for later

Comment (ilia-kebets-sonarsource) [reply test/integration/scanner.test.js:35]: remove only

**Review by saberduck — APPROVED** [review]

**Pull request merged**